# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditSubActionHandler.cpp
# Commit: 4b43fd9bf671
# Full Hash: 4b43fd9bf67144f7f66e7f7bd8561807cae9de77
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-02 21:27:32
# Regressor Bug: 1618089
# File Overlap Count: 1
# Description:
#   Bug 1618089 - part 1: Redsign `EditorDOMPointBase` with template methods r=m_kato
#   
#   Some methods of `EditorDOMPointBase` assumes the container node type is
#   `nsINode`.  However, it's not good for reuse.  Therefore, this patch makes
#   most methods of them be template methods.
# ==============================================================================

diff -r b511a045d72e -r 4b43fd9bf671 editor/libeditor/HTMLEditSubActionHandler.cpp
--- a/editor/libeditor/HTMLEditSubActionHandler.cpp	Mon Mar 02 09:52:58 2020 +0000
+++ b/editor/libeditor/HTMLEditSubActionHandler.cpp	Mon Mar 02 07:33:23 2020 +0000
@@ -3525,8 +3525,7 @@
         advanced,
         "Failed to advance offset to after child of rightBlock, "
         "leftBlock is a descendant of the child");
-    nsresult rv =
-        WSRunObject::Scrub(*this, EditorDOMPoint::AtEndOf(*leftBlock));
+    nsresult rv = WSRunObject::Scrub(*this, EditorDOMPoint::AtEndOf(leftBlock));
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return EditActionIgnored(rv);
     }
@@ -3552,7 +3551,7 @@
 
     // Do br adjustment.
     RefPtr<Element> invisibleBRElement =
-        GetInvisibleBRElementAt(EditorDOMPoint::AtEndOf(*leftBlock));
+        GetInvisibleBRElementAt(EditorDOMPoint::AtEndOf(leftBlock));
     EditActionResult ret(NS_OK);
     if (NS_WARN_IF(mergeListElements)) {
       // Since 2002, here was the following comment:
@@ -3767,7 +3766,7 @@
   }
   // Do br adjustment.
   RefPtr<Element> invisibleBRElement =
-      GetInvisibleBRElementAt(EditorDOMPoint::AtEndOf(*leftBlock));
+      GetInvisibleBRElementAt(EditorDOMPoint::AtEndOf(leftBlock));
   EditActionResult ret(NS_OK);
   if (mergeListElements ||
       leftBlock->NodeInfo()->NameAtom() == rightBlock->NodeInfo()->NameAtom()) {
@@ -10760,7 +10759,7 @@
         }
         createdListElement = CreateNodeWithTransaction(
             MOZ_KnownLive(*ULOrOLOrDLTagName),
-            EditorDOMPoint::AtEndOf(*targetDivElement));
+            EditorDOMPoint::AtEndOf(targetDivElement));
         if (NS_WARN_IF(Destroyed())) {
           return NS_ERROR_EDITOR_DESTROYED;
         }
@@ -10824,7 +10823,7 @@
         // XXX So, createdListElement may be set to a non-list element.
         createdListElement = CreateNodeWithTransaction(
             MOZ_KnownLive(*containerName),
-            EditorDOMPoint::AtEndOf(*targetDivElement));
+            EditorDOMPoint::AtEndOf(targetDivElement));
         if (NS_WARN_IF(Destroyed())) {
           return NS_ERROR_EDITOR_DESTROYED;
         }